src/Application.vala

```diff
@@ -216,15 +220,15 @@ public class Terminal.Application : Gtk.Application {
     protected override int command_line (ApplicationCommandLine command_line) {
         unowned var options = command_line.get_options_dict ();
         var window = (MainWindow) active_window;
-        bool new_window;
+        bool new_window = false;
+        options.lookup ("new-window", "b", out new_window);
+        /* Uncertain whether tabs should be restored when app is launched with working directory from commandline.
+         * Currently they are set to restore (subject to the restore-tabs setting).
+         * If it is desired that tabs should never be restored in these circimstances add another check below.
+         */
 
-        if (window == null || options.lookup ("new-window", "b", out new_window) && new_window) {
-            /* Uncertain whether tabs should be restored when app is launched with working directory from commandline.
-             * Currently they are set to restore (subject to the restore-tabs setting).
-             * If it is desired that tabs should never be restored in these circimstances add another check below.
-             */
-            bool restore_tabs = !("commandline" in options || "execute" in options) || window == null;
-            window = new MainWindow (this, restore_tabs);
+        if (window == null || new_window) {
+            window = new MainWindow (this, !new_window);
```

Comment:
if this is the first window, we aren't restoring. is that expected? i don't see a why `-n` won't restore the last session if it's the first window.

Reply:
I think is is cleaner and more predictable if `-n` always creates a fresh new window.  If the user wants restored tabs then omit the `-n`.

Reply:
keep in mind that `-n` will also mean a hard "drop tabs" when used in the first launch, because there's no way to not restore the tabs in any other case that isn't the first launch.

if you fine with that behaviour, i won't oppose the change.

Reply:
On reflection I have left restoring tabs to MainWindow and not dependent on options as it is simpler.